synfig-studio/src/gui/instance.cpp

```diff
@@ -223,25 +223,128 @@ Instance::set_redo_status(bool x)
 	signal_undo_redo_status_changed()();
 }
 
+bool
+studio::Instance::run_plugin_with_arguments(std::string plugin_path, const std::vector<std::string>& args)
```

Comment:
I'd split up this function in two:
* One detects a valid python binary filepath - and maybe build the full command line
* Other runs the script (open pipe and everything)

Reply:
I wanted to keep this merge request small so I haven't refactored much of the plugin code.

But I think it would also be a good idea to move most of the code in run_plugin into the plugin manager (or maybe even have plugin struct become a proper class and have it there).

`run_plugin_with_arguments` is really just extracting part of the code that used to be in `run_plugin`.